wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Create/Logic/GroupConversationCreationCoordinator.swift

```diff
@@ -0,0 +1,218 @@
+        NotificationCenter.default.removeObserver(self, name: ZMConversation.missingLegalHoldConsentNotificationName, object: nil)
+        NotificationCenter.default.removeObserver(self, name: ZMConversation.nonFederatingBackendsNotificationName, object: nil)
+        NotificationCenter.default.removeObserver(self, name: ZMConversation.unknownResponseErrorNotificationName, object: nil)
+        NotificationCenter.default.removeObserver(self, name: ZMConversation.insertedConversationUpdatedNotificationName, object: nil)
+        state = .ready
+    }
+
+    @objc func conversationCreatedHandler(notification: Notification) {
+        guard let conversation = extractConversation(notification: notification)
+            else { return }
+        eventHandler?(.hideLoader)
+        eventHandler?(.success(conversation: conversation))
+    }
+
+    @objc func unsupportedErrorsHandler(notification: Notification) {
+        guard extractConversation(notification: notification) != nil
+            else { return }
+        eventHandler?(.hideLoader)
+        eventHandler?(.failure(failureType: .other))
+    }
+
+    @objc func missingLegalConsentErrorHandler(notification: Notification) {
+        guard extractConversation(notification: notification) != nil
+            else { return }
+        typealias ConversationError = L10n.Localizable.Error.Conversation
+        eventHandler?(.hideLoader)
+        eventHandler?(
+            .presentPopup(
+                popupType: .missingLegalHoldConsent(
+                    completionHandler: { [weak self] in
+                        self?.eventHandler?(.failure(failureType: .missingLegalHoldConsent))
+                    }
+                )
+            )
+        )
+    }
+
+    @objc func nonFederatingBackendsErrorHandler(notification: Notification) {
+            guard extractConversation(notification: notification) != nil
+                else { return }
+        guard let userInfo = notification.userInfo,
+              userInfo.keys.contains(ZMConversation.UserInfoKeys.nonFederatingBackends.rawValue),
+              let nonFederatingBackends = userInfo[ZMConversation.UserInfoKeys.nonFederatingBackends.rawValue] as? NonFederatingBackendsTuple
+            else { return }
+        eventHandler?(.hideLoader)
+        eventHandler?(
+            .presentPopup(
+                popupType: .nonFederatingBackends(
+                    backends: nonFederatingBackends,
+                    actionHandler: { [weak self] action in
+                        switch action {
+                        case .editParticipantsList:
+                            self?.finalize()
+                        case .discardGroupCreation:
+                            self?.eventHandler?(.failure(failureType: .nonFederatingBackends))
+                            self?.finalize()
+                        case .learnMore:
+                            guard let url = URL(string: "https://support.wire.com") else { return }
```

Comment:
We don't usually hardcode links and we might already have this link. Please check` extension URL`